app/controllers/users_controller.rb

```diff
@@ -0,0 +1,25 @@
+# frozen_string_literal: true
+
+class UsersController < ApplicationController
+  def new
+    @user = User.new
+  end
+
+  def create
+    @user = User.new(user_params)
+    if @user.save
+      login(params[:email], params[:password])
+      flash[:success] = "Welcome! You have signed up successfully"
+      redirect_to root_path
+    else
+      flash[:error] = @user.errors.full_messages
```

Comment:
Usually for validation errors, you don't display them in a flash message but instead render them on the page using something like: 

```
   <% if @user.errors.full_messages.any? %>
      <%= @user.errors.full_messages.join(", ") %>
   <% end %>
```

or you can even print the individual messages out next to their fields,

something like...
```
   <%= @user.errors["email"] %>
```

which is usually used to add an `.has-error` css class or similar.

But this is ok! Just thought you might want to be aware of it.

Reply:
I was aware but it wasn't working